sridha2412/assignment1/string_anagrams.py

```diff
@@ -14,35 +14,31 @@ def isAnagram(str1, str2, caseSensitive):
 # assumption: sentence can only include alphabets and spaces
 # method to check if two sentences are anagrams
 # method takes two sentences and a boolean value. If true, check case sensitive anagrams else ignore case
-def isSentenceAnagram(str1, str2, caseSensitive):
-    # if ignoring case, convert both strings to lower case
-    if not caseSensitive:
-        str1 = str1.lower()
-        str2 = str2.lower()
-
+def is_sentence_anagram(str1, str2, caseSensitive):
     # split sentence into words
     str1 = str1.split(' ')
     str2 = str2.split(' ')
-
-    # sort each word and add it to list
-    for i, word in enumerate(str1):
-        str1[i] = sorted(list(word))
-    for i, word in enumerate(str2):
-        str2[i] = sorted(list(word))
-
-    # sort and compare the strings
-    return sorted(str1) == sorted(str2)
+    # check both have equal number of words
+    if len(str1) != len(str2):
+        return False
+    # iterate over both lists
+    for i in range(0, len(str1)):
```

Comment:
2 notes: 
- range(0, 10) is equivalent to range(10). When the first element is not set, then its value defaults to zero.
- this is a problematic change: the previous algorithm was O(n * ln (n)) but now it is O(n * m) where n is the number of words in the first sentences and is the number of words in the second sentence.

I suggest you create a function that takes a sentence and returns a structure whose equality method test for anagrams. This function can take the sentence as an input and return a list of occurences count dictionaries. Do not forget to sort the words of the sentence before creating the list.

I will send a solution of anagram today.